src/sentry/integrations/github/client.py

```diff
@@ -45,7 +47,53 @@ def search_repositories(self, query):
         return self.get("/search/repositories", params={"q": query})
 
     def get_assignees(self, repo):
-        return self.get(f"/repos/{repo}/assignees")
+        return self.get_with_pagination(f"/repos/{repo}/assignees")
+
+    def get_with_pagination(self, path, *args, **kwargs):
+        """
+        Github uses the Link header to provide pagination links. Github recommends using the provided link relations and not constructing our own URL.
+        https://docs.github.com/en/rest/guides/traversing-with-pagination
+        """
+        try:
+            with sentry_sdk.configure_scope() as scope:
+                parent_span_id = scope.span.span_id
+                trace_id = scope.span.trace_id
+        except AttributeError:
+            parent_span_id = None
+            trace_id = None
+
+        with sentry_sdk.start_transaction(
+            op=f"{self.integration_type}.http.pagination",
+            name=f"{self.integration_type}.http_response.pagination.{self.name}",
+            parent_span_id=parent_span_id,
+            trace_id=trace_id,
+            sampled=True,
+        ):
+            output = []
+            resp = self.get(path, params={"per_page": self.page_size})
+            output.extend(resp)
+            page_number = 1
+
+            def get_next_link(resp):
+                link = resp.headers.get("link")
+                if link is None:
+                    return None
+
+                # Should be a comma separated string of links
+                links = link.split(",")
+
+                for link in links:
+                    # If there is a 'next' link return the URL between the angle brackets, or None
+                    if 'rel="next"' in link:
+                        return link[link.find("<") + 1 : link.find(">")]
```

Comment:
this can probs just be `link[1 : link.find(">")]`

Reply:
@MeredithAnya I think its good to check for the index of `<`. It's not guaranteed to always be at index 0.

Reply:
@NisanthanNanthakumar when would it not be at 0?

Reply:
@MeredithAnya currently it should always be at 0, but that'll make an assumption that GitHub's API Link headers will never change its format. I don't think we should rely on 3rd parties to stay consistent. The current implementation doesn't add any overhead for this check.

Reply:
Agreed with Nisanthan here

Reply:
@NisanthanNanthakumar @scefali I'm fine with keeping the current implementation, but I'm not sure I find it any less likely of breaking if GH changes their format (i.e. what if they remove angle brackets all-together). But yeah fine to keep as is